core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -459,7 +459,7 @@ private[spark] class BlockManager(
     for (i <- 1 to MAX_ATTEMPTS) {
       try {
         // Synchronous and will throw an exception if we cannot connect.
-        shuffleClient.asInstanceOf[ExternalShuffleClient].registerWithShuffleServer(
+        shuffleClient.asInstanceOf[ExternalBlockStoreClient].registerWithShuffleServer(
```

Comment:
nit: is this `asInstanceOf` necessary?

Reply:
Yep it's necessary because only ExternalBlockStoreClient has this function.